src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -68,7 +68,7 @@ public void ProcessStart_UseShellExecute_OnLinux_ThrowsIfNoProgramInstalled()
             if (!s_allowedProgramsToRun.Any(program => IsProgramInstalled(program)))
             {
                 Console.WriteLine($"None of the following programs were installed on this machine: {string.Join(",", s_allowedProgramsToRun)}.");
-                Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = Environment.CurrentDirectory }));
+                Assert.Throws<Win32Exception>(() => { using (var p = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = Environment.CurrentDirectory })) { } });
```

Comment:
Also, if for some reason we really do want to call Dispose in the case where the test fails, we should instead do the Dispose outside of the Throws rather than inside it; otherwise, it could actually be the Dispose that's throwing the exception and causing a false negative on the test passing.